frontend/src/screens/CurrentGoalPage.js

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from 'react';
+import { View, Text, ScrollView, TouchableOpacity } from 'react-native';
+import styles from '../styles/CurrentGoalStyle';
+import useAuthFetch from '../utils/useAuthFetch';
+
+const CurrentGoalPage = ({ route, navigation }) => {
+    const { goalId } = route.params;
+    const [goal, setGoal] = useState(null);
+    const [habitMap, setHabitMap] = useState({});
+    const { fetchWithAuth, loading } = useAuthFetch();
+
+    useEffect(() => {
+        const fetchGoal = async () => {
+            try {
+                const data = await fetchWithAuth(`http://10.0.2.2:8080/api/goals/${goalId}`); //emulator

```

Comment:
It is possible to add "http://10.0.2.2:8080/api" string in a variable set in a different file (environment_variables.js or something like that) and use this variable in the following way: API + '/goals/${goalId}'? If so, do this for all pages where you are doing http requests.